server/player.lua

```diff
@@ -390,32 +390,40 @@ function CheckPlayerData(source, playerData)
         InstalledApps = {},
     }
     local jobs, gangs = storage.fetchPlayerGroups(playerData.citizenid)
-    -- Job
-    if playerData.job and playerData.job.name and not GetJob(playerData.job.name) then playerData.job = nil end
-    playerData.job = playerData.job or {}
-    playerData.job.name = playerData.job.name or 'unemployed'
-    playerData.job.label = playerData.job.label or 'Civilian'
-    playerData.job.payment = playerData.job.payment or 10
-    playerData.job.type = playerData.job.type or 'none'
-    if QBX.Shared.ForceJobDefaultDutyAtLogin or playerData.job.onduty == nil then
-        playerData.job.onduty = GetJob(playerData.job.name).defaultDuty
-    end
-    playerData.job.isboss = playerData.job.isboss or false
-    playerData.job.grade = playerData.job.grade or {}
-    playerData.job.grade.name = playerData.job.grade.name or 'Freelancer'
-    playerData.job.grade.level = playerData.job.grade.level or 0
+
+    local job = GetJob(playerData.job?.name) or GetJob('unemployed')
+    if not job then
+        error("unemployed job not found. Is it in your config and/or database?")
+    end
+    local jobGrade = GetJob(playerData.job?.name) and playerData.job.grade.level or 0
```

Comment:
GetJob is redundant here and job grade should be grabbed from the `job` variable

Reply:
```suggestion
    local jobGrade = job.grade[playerData.job?.grade.level].level or 0
```
I believe this is what Berkie means. This should cover cases where a grade is deleted and the player has a grade that no longer exists

Reply:
the job variable is the actual job itself, not the player's job. Therefore it doesn't have player specific grade info, but rather all the grades of the job instead.

Reply:
> I believe this is what Berkie means. This should cover cases where a grade is deleted and the player has a grade that no longer exists

The reason why this won't work (aside from some off var names) is that job is either the saved job, or unemployed. If it's unemployed we want to disregard the saved player grade and go with 0.

Reply:
> > I believe this is what Berkie means. This should cover cases where a grade is deleted and the player has a grade that no longer exists
> 
> The reason why this won't work (aside from some off var names) is that job is either the saved job, or unemployed. If it's unemployed we want to disregard the saved player grade and go with 0.

My bad. I thought we were assembling the client side. disregard my comment